### DISCUSSION TITLE
How to nest blocks with text?

### DISCUSSION THREAD

**Discussion by bbsclient** [mermaid-js/mermaid#5502] [Q&A]
I am trying to create a block that contains text along with multiple child blocks.  By default the text of the parent appears behind the child blocks.  What is the best way to get the parent text to appear above the children.

block-beta
  block:Parent["Parent"]
    child1 child2
  end

I was able to work around it slightly by adding spaces to force multiple rows of children but I wasn't able to get the parent text to be top justified using classDef with vertical-align:top.

**Comment by CoryLR** [comment]
I have this need as well. I'm trying to use the `block-beta` diagram type make something that looks roughly like this (image attached) - nested blocks with grouping labels. I'm using Mermaid version 11.4.0 via the `bierner.markdown-mermaid` VS Code extension.
<img width="336" alt="nested-block-with-text-example" src="https://github.com/user-attachments/assets/588d9c22-1b52-4953-9423-69e44ea22236" />

Anyone know how to do something like this?

**Reply by roke-julian-lockhart** [reply]
> [`#discussioncomment-12104251`](https://github.com/orgs/mermaid-js/discussions/5502#discussioncomment-12104251)

@CoryLR, I've tried CommonMark and HTML. Both render as the undermentioned describes:

>  By default the text of the parent appears behind the child blocks.

I think that this needs to be reported as a bug. Do you or @bbsclient want to, or should I? I can't really be bothered to - feeling a little ill. :<

**Reply by CoryLR** [reply]
I wouldn’t mind, I could basically copy-paste what I commended above and link this thread. Think I should open it as an issue on the repo? @RokeJulianLockhart

**Reply by roke-julian-lockhart** [reply]
> [`#discussioncomment-12104711`](https://github.com/orgs/mermaid-js/discussions/5502#discussioncomment-12104711)

@CoryLR, yeah. Post the link here when you're done, please!
